neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -137,6 +143,20 @@ type VirtualMachineSpec struct {
 	// +kubebuilder:default:=true
 	// +optional
 	EnableSSH *bool `json:"enableSSH,omitempty"`
+
+	// OvercommitFactor sets a factor by which to discount the resource usage from this VM.
+	// For example, if all VMs have an OvercommitFactor of 2.0, then twice as many VMs of the same
+	// size could be scheduled onto nodes of the same size.
+	//
+	// OvercommitFactor is INHERENTLY DANGEROUS. Setting this value greater than 1 ONLY makes sense
+	// when all your VMs are much smaller than the nodes they run on, AND you have verified that
+	// average resource usage is, in practice, significantly below what's allocated.
+	//
+	// +kubebuilder:validation:Minimum=0.5
+	// +kubebuilder:validation:Maximum=2.0
+	// +kubebuilder:validation:ExclusiveMaximum=false
+	// +optional
+	OvercommitFactor *float64 `json:"overcommitFactor,omitempty"` // validations are just for reasonable values.
```

Comment:
Two points:
1. AFAIU disk is not currently accounted, but is overcommited. If all the tenants decide to use all their 20 GB of temp space (+LFC+swap) we will run out of disk. 
2. We should consider introducing separate `OvercommitCPU`, `OvercommitMemory`, `OvercommitDisk`. The previous point implies we consider overcommit on disk less dangerous.

Reply:
1. Yep
2. I don't see the benefit we'd get by adding this — least for right now. Our nodes are scaled with the same CPU:Memory ratio as our computes — separately overcommitting CPU and memory only provides more precise configuration if the space available to computes doesn't maintain the same ratio as compute units. Otherwise you can, in effect, get the same behavior just by setting `OvercommitFactor := min(OvercommitCPU, OvercommitMemory)`. Regarding `OvercommitDisk`, I agree that we should have this if/when we have disk accounting, but without disk accounting I don't see what the purpose would be?

Reply:
Thought about this again - changed my mind, I think you're right here. Currently our nodes use more of their memory than CPU, and having this would allow us to make changes to node types that reflect that.

Reply:
Implemented in 3bd0c2190207ece4b0d26c3eca91dfbdec98ebce